articles/150_ros_command_line_arguments.md

```diff
@@ -255,3 +321,50 @@ This signficantly increases command line verbosity, but still avoids the need fo
 Remove the need for double dash tokens (`--`), conventionally used to signify the end of CLI options for a command, by adding the `--ros-` prefix to all ROS specific command line flags e.g. `--ros-remap`, `--ros-param`, etc.
 In exchange, it makes argument extraction slightly more difficult as all options must be known ahead of time, whereas `--ros-args`-based namespacing can achieve the same with a couple rules.
 It also increases command line verbosity.
+
+## Precedence order discussion
+
+There are some issues of the current precedence rules:
+
+* Precedence orders for remapping rules and parameter assignments are inconsistent.
+* In the past, inconsistencies were detected between `rclcpp` and `rclpy` ([example issue](https://github.com/ros2/rclcpp/issues/953)).
+* Weird interaction with launch files.
+
+Example of the last item:
+
+```python
+LaunchDescription([
+  Node(
+    package='my_pkg',
+    exec='my_exec',
+    name='my_node',
+    parameters=['/path/to/my_params.yaml', {'my_int': '3'}]
+  )
+])
+```
+
+Where `my_params.yaml` can be find in *Precedence order of command line arguments* section.
+The parameter file is assigning `my_int` to `2`.
+Launch cannot know the fully qualified name of the node at launch time, as the node namespace wasn't specified.
+Thus, it needs to use a "wildcard" rule for the assignment made in the dictionary, and it won't override the parameter file assignment.
+
+Before https://github.com/ros2/launch_ros/pull/154 was merged, it was impossible to override a parameter assignment in a parameter file that targeted an specific node from a launch file.
+Now it's possible by doing:
+
+```python
+LaunchDescription([
+  Node(
+    package='my_pkg',
+    exec='my_exec',
+    name='my_node',
+    namespace='my_ns',
+    parameters=['/path/to/my_params.yaml', {'my_int': '3'}]
+  )
+])
+```
+
+As the fully qualified node name is known at launch time in this last case, `my_int` will be set to `3`.
+
+Possible solutions:
+* Warn when parameter files and parameter dictionaries are combined in `Node`, and the fully qualified node name is not known.
+* Modify precedence between wildcard and not wildcard assignments.
```

Comment:
@ivanpauno shouldn't this design doc weigh pros and cons and make an executive decision?

Reply:
as commented previously, I'm not trying to weight advantages and disadvantages at the moment.